src/pip/_internal/cli/cmdoptions.py

```diff
@@ -920,6 +922,35 @@ def check_list_path_option(options):
 )  # type: Callable[..., Option]
 
 
+discovered_auth_plugins = {
+    entry_point.name: entry_point
+    for entry_point
+    in iter_entry_points('pip.auth_plugins')
+}
+
+
+def _handle_auth_plugin(option, opt_str, value, parser):
+    # type: (Option, str, str, OptionParser) -> None
+    plugin = discovered_auth_plugins[value].load()
+    setattr(parser.values, option.dest, plugin.Auth)
+
+
+auth_plugin = partial(
+    PipOption,
+    '--auth', '--auth-plugin',
```

Comment:
As an user, I would find this to be confusing.

Reply:
Not sure what's confusing about this. It's quite common for options to have multiple names.

Reply:
I find `auth` and `auth-plugin` to have different first-impression meaning, but I don't represent all users so no worries, stick to what you think is best.

Reply:
Thanks for clarifying. It's similar to the '--src' option, which has three longer names: '--source', '--source-dir', '--source-directory'. The longer the name - the more informative and clear it is. Let's see if others find it confusing, too.
@McSinyx Don't worry. Your thoughts and perceptions are welcome. Thanks for thinking with me.